roles/tests-integration/lib/sniffer.rs

```diff
@@ -58,19 +58,79 @@ pub struct Sniffer {
     messages_from_downstream: MessagesAggregator,
     messages_from_upstream: MessagesAggregator,
     check_on_drop: bool,
-    intercept_messages: Vec<InterceptMessage>,
+    action: Option<InterceptAction>,
+}
+
+/// Represents an action that [`Sniffer`] can take on intercepted messages.
+#[derive(Debug, Clone)]
+pub enum InterceptAction {
+    /// Blocks the message stream after encountering a specific message.
+    IgnoreFromMessage(IgnoreFromMessage),
+    /// Intercepts and modifies a message before forwarding it.
+    ReplaceMessage(Box<ReplaceMessage>),
+}
+
+impl InterceptAction {
+    /// Returns the action if it is `IgnoreFromMessage` or `ReplaceMessage`  
+    /// with the specified message type.
+    pub fn find_matching_action(
```

Comment:
Maybe something like the following to reduce duplicated code(and also receive `&self` instead of `Option<action>`, otherwise why would you call this function with a `None` value?):

```suggestion
    pub fn find_matching_action(
    &self,
    msg_type: MsgType,
    direction: MessageDirection,
) -> Option<InterceptAction> {
    match self {
        InterceptAction::IgnoreFromMessage(bm)
        | InterceptAction::ReplaceMessage(bm)
            if bm.direction == direction && bm.expected_message_type == msg_type =>
        {
            Some(action)
        }
        _ => None,
    }
}
```